Can't install matplotlib from source due to recent addition of QHULL_LIB_CHECK to src/qhull_wrap.c

I'm currently trying to use pip (on a mac running osx El Capitan) to install a particular commit of matplotlib:
```
pip install -U --upgrade-strategy only-if-needed git+https://github.com/matplotlib/matplotlib.git@91e74257b770d#egg=matplotlib
```

I need this particular commit so I can fix an issue I (and others) are having with the width of type 42 fonts in v2.0. Unfortunately, this is failing with:
```
...

    building 'matplotlib._qhull' extension
    clang -Wno-unused-result -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -I/usr/local/include -I/usr/local/opt/sqlite/include -DMPL_DEVNULL=/dev/null -DPY_ARRAY_UNIQUE_SYMBOL=MPL_matplotlib__qhull_ARRAY_API -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -D__STDC_FORMAT_MACROS=1 -I/usr/local/lib/python3.4/site-packages/numpy/core/include -I/usr/local/include -I/usr/include -I/usr/X11/include -I/opt/X11/include -I. -I/usr/local/Cellar/python3/3.4.2_1/Frameworks/Python.framework/Versions/3.4/include/python3.4m -c src/qhull_wrap.c -o build/temp.macosx-10.9-x86_64-3.4/src/qhull_wrap.o
    src/qhull_wrap.c:112:5: error: use of undeclared identifier 'QHULL_LIB_CHECK'
        QHULL_LIB_CHECK
        ^
    1 error generated.
    error: command 'clang' failed with exit status 1

...
```

I know matplotlib is finding the relevant qhull library:
```
...

REQUIRED DEPENDENCIES AND EXTENSIONS

...

                     qhull: yes [Using system Qhull (version unknown, no pkg-
                            config info)]

...
```

so I'm not sure what the problem is. Examining `qhull_wrap.c` in the repo shows that the offending line was recently added in a recent commit with the note `Update qhull to 2015.2`. Is the fix that I should try to upgrade the qhull libs on my system? The [matplotlib install doc](http://matplotlib.org/users/installing.html) makes it sound like the qhull libs are included with matplotlib, so I'm somewhat confused.

--- a/setupext.py
+++ b/setupext.py
@@ -1033,7 +1033,7 @@ class LibAgg(SetupPackage):
         if self.found_external:
             pkg_config.setup_extension(ext, 'libagg')
         else:
-            ext.include_dirs.append('extern/agg24-svn/include')
+            ext.include_dirs.insert(0, 'extern/agg24-svn/include')
             if add_sources:
                 agg_sources = [
                     'agg_bezier_arc.cpp',
@@ -1348,23 +1348,15 @@ class Qhull(SetupPackage):
                 'libqhull', 'libqhull/qhull_a.h', min_version='2015.2')
         except CheckFailed as e:
             self.__class__.found_pkgconfig = False
-            # Qhull may not be in the pkg-config system but may still be
-            # present on this system, so check if the header files can be
-            # found.
-            include_dirs = [
-                os.path.join(x, 'libqhull') for x in get_include_dirs()]
-            if has_include_file(include_dirs, 'qhull_a.h'):
-                return 'Using system Qhull (version unknown, no pkg-config info)'
-            else:
-                self.__class__.found_external = False
-                return str(e) + ' Using local copy.'
+            self.__class__.found_external = False
+            return str(e) + ' Using local copy.'
 
     def add_flags(self, ext):
         if self.found_external:
             pkg_config.setup_extension(ext, 'qhull',
                                        default_libraries=['qhull'])
         else:
-            ext.include_dirs.append('extern')
+            ext.include_dirs.insert(0, 'extern')
             ext.sources.extend(sorted(glob.glob('extern/libqhull/*.c')))
 
 
@@ -1380,7 +1372,7 @@ class TTConv(SetupPackage):
             ]
         ext = make_extension('matplotlib.ttconv', sources)
         Numpy().add_flags(ext)
-        ext.include_dirs.append('extern')
+        ext.include_dirs.insert(0, 'extern')
         return ext
 
 
@@ -1536,7 +1528,7 @@ class BackendTkAgg(OptionalBackendPackage):
         return ext
 
     def add_flags(self, ext):
-        ext.include_dirs.extend(['src'])
+        ext.include_dirs.insert(0, 'src')
         if sys.platform == 'win32':
             # PSAPI library needed for finding Tcl / Tk at run time
             ext.libraries.extend(['psapi'])

